src/core/meshcop/timestamp.hpp

```diff
@@ -143,16 +167,29 @@ class Timestamp : public Clearable<Timestamp>
      */
     static int Compare(const Timestamp *aFirst, const Timestamp *aSecond);
 
+    /**
+     * This static method compares two timestamps.
+     *
+     * @param[in]  aFirst   A reference to the first timestamp to compare.
+     * @param[in]  aSecond  A reference to the second timestamp to compare.
+     *
+     * @retval -1  if @p aFirst is less than @p aSecond (`aFirst < aSecond`).
+     * @retval  0  if @p aFirst is equal to @p aSecond (`aFirst == aSecond`).
+     * @retval  1  if @p aFirst is greater than @p aSecond (`aFirst > aSecond`).
+     *
+     */
+    static int Compare(const Timestamp &aFirst, const Timestamp &aSecond);
+
 private:
+    static constexpr uint8_t  kSecondsOffset       = 16;
+    static constexpr uint64_t kSecondsMask         = 0xffffffffffffULL << kSecondsOffset;
     static constexpr uint8_t  kTicksOffset         = 1;
-    static constexpr uint16_t kTicksMask           = 0x7fff << kTicksOffset;
+    static constexpr uint64_t kTicksMask           = 0x7fffULL << kTicksOffset;
     static constexpr uint16_t kMaxRandomTicks      = 0x7fff;
     static constexpr uint8_t  kAuthoritativeOffset = 0;
-    static constexpr uint16_t kAuthoritativeMask   = 1 << kAuthoritativeOffset;
+    static constexpr uint64_t kAuthoritativeMask   = 1ULL << kAuthoritativeOffset;
 
-    uint16_t mSeconds16; // bits 32-47
-    uint32_t mSeconds32; // bits 0-31
-    uint16_t mTicks;
+    uint64_t mValue;
```

Comment:
Noticed this PR is also changing the OT internal `Timestamp` definition to use `uint64_t`. Any reason for this change? 


If it is jut an improvement and not needed for this PR (which if I understand it correctly it seems to be) I suggest adding it in a a separate PR. 

I suggesting keeping this PR focused on changing the public API to provide more detail about time stamp. Thoughts?